flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendListTest.java

```diff
@@ -28,21 +28,100 @@
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
-import org.mockito.Mockito;
 
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 /** Tests for the LIST command. */
 class CliFrontendListTest extends CliFrontendTestBase {
 
+    private static final List<Long> TRICKY_START_TIMES =
+            Arrays.asList(
+                    1664177946934L,
```

Comment:
Can we make this more focused? For this particular issue we should require exactly 2 timestamps.

Reply:
This issue requires at least 32 items to occur. I do not like this either, though.

Reply:
There were another similar issue I found: https://github.com/apache/flink/pull/5878/files

Reply:
I played a bit with the test data (randomly), but it passes pretty quick, so I just included the whole faulty dataset as is.

Reply:
From what I found you need 32 elements for the sort to _fail with an exception_, but you can use less elements and end up with an unsorted list.

Reply:
That is true, but with the current implementation you cannot verify that from the test. The sorting happens in a `private static` method, which just prints the result out. Is it worth it to change that code for this and make it more testable? Or since the fix itself is kinda trivial just leave the test case as is? What do you think?

Reply:
Should be easy enough to modify the code to make the testing easier.

```

    private static void printJobStatusMessages(List<JobStatusMessage> jobs) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
        sortJobStatusMessages(jobs)
                .forEachOrdered(
                        job ->
                                System.out.println(
                                        dateFormat.format(new Date(job.getStartTime()))
                                                + " : "
                                                + job.getJobId()
                                                + " : "
                                                + job.getJobName()
                                                + " ("
                                                + job.getJobState()
                                                + ")"));
    }

    @VisibleForTesting
    static Stream<JobStatusMessage> sortJobStatusMessages(List<JobStatusMessage> jobs) {
        Comparator<JobStatusMessage> startTimeComparator =
                (o1, o2) -> (int) (o1.getStartTime() - o2.getStartTime());
        Comparator<Map.Entry<JobStatus, List<JobStatusMessage>>> statusComparator =
                (o1, o2) ->
                        String.CASE_INSENSITIVE_ORDER.compare(
                                o1.getKey().toString(), o2.getKey().toString());

        Map<JobStatus, List<JobStatusMessage>> jobsByState =
                jobs.stream().collect(Collectors.groupingBy(JobStatusMessage::getJobState));
        return jobsByState.entrySet().stream()
                .sorted(statusComparator)
                .map(Map.Entry::getValue)
                .flatMap(List::stream)
                .sorted(startTimeComparator);
    }
```

Reply:
This would also allow us to test the grouping-by-state behavior, which I'm not sure of if it even works properly.

Reply:
Yeah, I thought something like that by changing the code, will update the PR in a bit.